src/state/MediaViewModel.ts

```diff
@@ -433,20 +437,35 @@ abstract class BaseUserMediaViewModel extends BaseMediaViewModel {
  * The local participant's user media.
  */
 export class LocalUserMediaViewModel extends BaseUserMediaViewModel {
+  /**
+   * The local video track as an observable that emits whenever the track
+   * changes, the camera is switched, or the track is muted.
+   */
+  private readonly videoTrack$: Observable<LocalVideoTrack | null> =
+    this.video$.pipe(
+      switchMap((v) => {
+        const track = v?.publication?.track;
+        if (!(track instanceof LocalVideoTrack)) return of(null);
+        return merge(
+          // Watch for track restarts because they indicate a camera switch
+          fromEvent(track, TrackEvent.Restarted).pipe(
+            startWith(null),
+            map(() => track),
+          ),
+          fromEvent(track, TrackEvent.Muted).pipe(map(() => null)),
```

Comment:
```suggestion
          // Emits only when Muted instead of emitting on mute state changes. (on un-mute it will not emit)
          // We set the track to null here. On un-mute, the `fromEvent(..., TrackEvent.Restarted)` will make sure that `videoTrack$` is set to `track` again.
          fromEvent(track, TrackEvent.Muted).pipe(map(() => null)),
```

Do I understand this correctly?

Reply:
Yes - commented.